src/diag/diag_process.cpp

```diff
@@ -354,6 +354,8 @@ void Diag::DiagReceiveDone(RadioPacket *aFrame, ThreadError aError)
 
         sStats.received_packets++;
     }
+
+    otPlatRadioReceive(sChannel);
```

Comment:
Reenter receive mode after each receive done, in case some radio platforms don't take receive mode as default.